### PR TITLE
Feat: Entity Detection and linked with Bix Attack

### PR DESCRIPTION
**Pull request by UriMurcia**
Entity detection is in charge of detecting which enemy is in front of the player in order to perform attacks to them.
It consists on a sphere detecting all enemies near the player and an angle detecting which of them are in front of the player. The enemy selected will be, for the moment, the nearest to the player inside this angle. For now, the player only hits one enemy at the time.
I've also linked this script with the BixAttack script in order to hit the enemy selected.

To test this you can go to the GameplayFullTrial_VS3.axolotl

### CODE REVIEW

**Comment by markmeliaa** [comment]
The gameplay scene was outdated and not working after merging #557 and had to update it. If you bring game-programming into your branch it should fix that (and the conflict).

**Review by markmeliaa — APPROVED** [review]
Everything looks and works correctly, great job Uri!

**Comment by UriMurcia** [comment]
> Everything looks and works correctly, great job Uri!

Okaay, thank you Mark! I merged game-programming and updated the scene :)

**Review by AuxPort66 — APPROVED** [review]

**Review by DADSILVER — APPROVED** [review]
Good job, Only 1 question. Should not All references to debug draw be only in the engine?

File: Game/Scripts/BixAttackScript.cpp [hunk 1]
```diff
@@ -116,43 +79,23 @@ void BixAttackScript::PerformAttack()
...
-	for (const Ray& ray : rays)
+	GameObject* enemyAttacked = enemyDetection->GetEnemySelected();
+
+	if (enemyAttacked != nullptr)
 	{
-		LineSegment line(ray, rayAttackSize);
-		RaycastHit hit;
-		if (Physics::Raycast(line, hit, transform->GetOwner()))
+		HealthSystem* healthScript = enemyAttacked->GetComponent<HealthSystem>();
+		float damageAttack = playerManager->GetPlayerAttack();
```
[the remark this answers is not in the document]
Comment (UriMurcia) [reply Game/Scripts/BixAttackScript.cpp:87]: well, it's just for cleaning code, I don't think it's a problem ^^

**Pull request merged**